From ac042485f4a7cc325e562935eacdbe36b848a23e Mon Sep 17 00:00:00 2001
From: Chris Zhong <zyw@rock-chips.com>
Date: Tue, 28 Nov 2017 19:47:24 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix compile warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

fix this compile warning:
drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c:3518:31:
warning: comparison of constant ‘3’ with boolean expression is alwa
false [-Wbool-compare] error, forbidden warning: rtw_p2p.c:3518

Change-Id: Ic900425434f3ae0069237bbe6d3cb8242e31ed0f
Signed-off-by: Chris Zhong <zyw@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c   | 4 ++--
 drivers/net/wireless/rockchip_wlan/rtl8723au/core/rtw_p2p.c   | 4 ++--
 .../net/wireless/rockchip_wlan/rtl8723bs-vq0/core/rtw_p2p.c   | 4 ++--
 drivers/net/wireless/rockchip_wlan/rtl8812au/core/rtw_p2p.c   | 4 ++--
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c b/drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c
index d96d00f9401e..1add6b6b2490 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8189fs/core/rtw_p2p.c
@@ -3515,7 +3515,7 @@ _func_enter_;
 
 					rtw_p2p_set_state(pwdinfo, P2P_STATE_LISTEN);
 					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&
-					!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					   (pmlmeinfo->state & 0x03) != WIFI_FW_AP_STATE)
 					{
 						val8 = 1;
 						rtw_hal_set_hwreg(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
@@ -3539,7 +3539,7 @@ _func_enter_;
 				if ( pbuddy_mlmeext->cur_channel != pwdinfo->listen_channel )
 				{
 					set_channel_bwmode(padapter, pbuddy_mlmeext->cur_channel, pbuddy_mlmeext->cur_ch_offset, pbuddy_mlmeext->cur_bwmode);
-					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) && (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{						
 						val8 = 0;
 						rtw_hal_set_hwreg(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723au/core/rtw_p2p.c b/drivers/net/wireless/rockchip_wlan/rtl8723au/core/rtw_p2p.c
index dcb7dc75e6cc..5437afdbbe17 100755
--- a/drivers/net/wireless/rockchip_wlan/rtl8723au/core/rtw_p2p.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723au/core/rtw_p2p.c
@@ -3475,7 +3475,7 @@ _func_enter_;
 
 					rtw_p2p_set_state(pwdinfo, P2P_STATE_LISTEN);
 					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&
-					!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					   (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{	
 						val8 = 1;
 						rtw_hal_set_hwreg(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
@@ -3500,7 +3500,7 @@ _func_enter_;
 				{
 					set_channel_bwmode(padapter, pbuddy_mlmeext->cur_channel, pbuddy_mlmeext->cur_ch_offset, pbuddy_mlmeext->cur_bwmode);
 					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&	
-					!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					   (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{
 						val8 = 0;
 						padapter->HalFunc.SetHwRegHandler(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723bs-vq0/core/rtw_p2p.c b/drivers/net/wireless/rockchip_wlan/rtl8723bs-vq0/core/rtw_p2p.c
index 5f2f99c0a2cd..053256fec9b9 100755
--- a/drivers/net/wireless/rockchip_wlan/rtl8723bs-vq0/core/rtw_p2p.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723bs-vq0/core/rtw_p2p.c
@@ -3504,7 +3504,7 @@ _func_enter_;
 
 					rtw_p2p_set_state(pwdinfo, P2P_STATE_LISTEN);
 					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&
-					!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					   (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{
 						val8 = 1;
 						rtw_hal_set_hwreg(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
@@ -3528,7 +3528,7 @@ _func_enter_;
 				if ( pbuddy_mlmeext->cur_channel != pwdinfo->listen_channel )
 				{
 					set_channel_bwmode(padapter, pbuddy_mlmeext->cur_channel, pbuddy_mlmeext->cur_ch_offset, pbuddy_mlmeext->cur_bwmode);
-					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) && (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{						
 						val8 = 0;
 						padapter->HalFunc.SetHwRegHandler(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8812au/core/rtw_p2p.c b/drivers/net/wireless/rockchip_wlan/rtl8812au/core/rtw_p2p.c
index 5f2f99c0a2cd..053256fec9b9 100755
--- a/drivers/net/wireless/rockchip_wlan/rtl8812au/core/rtw_p2p.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8812au/core/rtw_p2p.c
@@ -3504,7 +3504,7 @@ _func_enter_;
 
 					rtw_p2p_set_state(pwdinfo, P2P_STATE_LISTEN);
 					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&
-					!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					   (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{
 						val8 = 1;
 						rtw_hal_set_hwreg(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
@@ -3528,7 +3528,7 @@ _func_enter_;
 				if ( pbuddy_mlmeext->cur_channel != pwdinfo->listen_channel )
 				{
 					set_channel_bwmode(padapter, pbuddy_mlmeext->cur_channel, pbuddy_mlmeext->cur_ch_offset, pbuddy_mlmeext->cur_bwmode);
-					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) &&!(pmlmeinfo->state&0x03) == WIFI_FW_AP_STATE)
+					if(!check_buddy_mlmeinfo_state(padapter, WIFI_FW_AP_STATE) && (pmlmeinfo->state&0x03) != WIFI_FW_AP_STATE)
 					{						
 						val8 = 0;
 						padapter->HalFunc.SetHwRegHandler(padapter, HW_VAR_MLME_SITESURVEY, (u8 *)(&val8));
-- 
2.35.3

